### PR TITLE
Create Prims_Algorithm.kt

### LINKED ISSUE #2201: Prims Algorithm
- [x] C
- [x] C++
- [ ] C# - @ritish099
- [ ] CoffeeScript
- [ ] Dart - @Swarnimashukla
- [ ] Go - @adi44
- [x] Java
- [x] JS - @sukritishah15
- [x] Kotlin - @darkmatter18
- [ ] Php - @prince-09
- [x] Python
- [x] Readme
- [ ] Ruby
- [ ] Rust
- [ ] TypeScript - @BhuwanChandra

### PR DESCRIPTION
**Pull request by darkmatter18**
Fixes #2201 

#### Checklist:
- [x] 4 space indentation.
- [x] Coding conventions are followed.
- [x] Input is taken dynamically.
- [x] Sample Input / Output is added at the end of file.
- [x] Logic Documentation (Comments).
- [x] File names are correct.

Make sure these boxes are checked before your pull request (PR) is ready to be reviewed and merged. Thanks!

#### Changes proposed in this pull request:
- Added Prims_Algorithm.kt

#### Languages Used:
- Kotlin

#### Files Added:
- Prims_Algorithm.kt

> We're happy to help you get this ready -- don't be afraid to ask for help.

Thanks!

### CODE REVIEW

**Comment by darkmatter18** [comment]
@jainaman224 Please look at it!

File: Prims_Algorithm/Prims_Algorithm.kt [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+    /**
+     * Function to compute MST
+     *
+     * @param graph
+     * @return mst
+     */
+    private fun prims(graph: ArrayList<ArrayList<Edge?>>): ArrayList<Edge?> { // ArrayList to store obtained MST
+        val mst = ArrayList<Edge?>()
+        // Priority Queue which gives minimum weight edge
+        val pq = PriorityQueue(Comparator { edge1: Edge?, edge2: Edge? -> if (edge1!!.weight < edge2!!.weight) return@Comparator -1 else if (edge1.weight > edge2.weight) return@Comparator 1 else return@Comparator 0 })
```
Comment (asha15) [inline Prims_Algorithm/Prims_Algorithm.kt:16]: one line should contain max 80 characters. please break this line

File: Prims_Algorithm/Prims_Algorithm.kt [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+        // add the source vertex in Priority queue
+        pq.addAll(graph[0])
+        var cost = 0.0
+        val size = graph.size
+        val visited = BooleanArray(size)
+        // mark source vertex as visited
+        visited[0] = true
```
Comment (asha15) [inline Prims_Algorithm/Prims_Algorithm.kt:23]: keep a line space after this

File: Prims_Algorithm/Prims_Algorithm.kt [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+        // Keep extracting edges till priority is not empty
+        while (!pq.isEmpty()) { // extract edge with minimum weight from priority queue
+            val edge = pq.peek()
+            // pop the edge from priority queue
+            pq.poll()
+            if (visited[edge!!.to] && visited[edge.from]) continue
+            // mark the from vertex as visited
+            visited[edge.from] = true
+            for (edge1 in graph[edge.to]) { // if the adjacent vertex to source is unvisited, add the edge to the priority queue
```
Comment (asha15) [inline Prims_Algorithm/Prims_Algorithm.kt:32]: take the comment into the previous line

**Comment by darkmatter18** [comment]
@asha15 @jainaman224 Please Check it

**Comment by asha15** [comment]
Good job :)
please squash your commits and let me know

**Comment by darkmatter18** [comment]
@asha15 Done! Kindly check

**Comment by asha15** [comment]
nope. you still have four commits.
please make it as one https://www.freecodecamp.org/forum/t/how-to-squash-multiple-commits-into-one-with-git-squash/13231

**Comment by darkmatter18** [comment]
@asha15 I think it is done now. Kindly check.
And Please assign me #2555 , #2480 , #2481 , #2477 for Kotlin language

**Pull request merged**